lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -1969,6 +1969,31 @@ static SILValue getTensorProtocolHandleMember(SILValue v, SILLocation loc,
   return v;
 }
 
+/// Given a type, recursively collect all innermost element types if it's an
+/// aggregate of TensorHandle's or dtypes.
+static void collectInnermostTensorFlowDTypes(
```

Comment:
Actually, this should return a `bool`, otherwise applying this on an aggregate type that has non-tensors will crash. Could you make this return `true` if it successfully collects all dtype, and `false` otherwise?